src/index.js

```diff
@@ -87,15 +86,14 @@ export type OptionsData = {
   validationRules?: ?Array<(ValidationContext) => ASTVisitor>,
 
   /**
-   * An optional compatible associative object to cache the results of validation.
+   * An optional function to wrap validation operations.
    */
-  validationCache?: {|
-    set: (
-      key: string,
-      value: $ReadOnlyArray<GraphQLError>,
-    ) => $ReadOnlyArray<GraphQLError>,
-    get: (key: string) => $ReadOnlyArray<GraphQLError>,
-  |},
+  validationWrapper?: ?(
```

Comment:
@jamesmoriarty Thanks for the effort 👍 

> Unsure about the best method signature. Should it pass in the response? Would you want the same for execute?

Looking more into the source code I notice we already use the same approach for `formatError`:
https://github.com/graphql/express-graphql/blob/7ab2bc958a2184fe35fb6cd137f369b5b3877dec/src/index.js#L75-L80
https://github.com/graphql/express-graphql/blob/7ab2bc958a2184fe35fb6cd137f369b5b3877dec/src/index.js#L336-L338

I think we should duplicate this approach for both `validate` and `execute`.
Since `graphql` is a peer dependency you need to add it into your `package.json` anyway so it mean you can just write:
```js
const { validate } = require('graphql');

app.post('/graphql', graphqlHTTP({
  schema: MyGraphQLSchema,
  graphiql: false
  validate(schema, documentAST, validationRules) {
    // some caching logic
    cache[hash] = validate(schema, documentAST, validationRules);
    // more caching logic
    return cache[hash]
  },
}));
```

Reply:
Looks great. Will refactor towards this.